ext/bcmath/bcmath.c

```diff
@@ -643,6 +643,96 @@ PHP_FUNCTION(bccomp)
 }
 /* }}} */
 
+/* {{{ floor or ceil */
+static void bcfloor_or_bcceil(INTERNAL_FUNCTION_PARAMETERS, bool is_floor)
+{
+	zend_string *numstr;
+	bc_num num, result;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(numstr)
+	ZEND_PARSE_PARAMETERS_END();
+
+	bc_init_num(&num);
+	bc_init_num(&result);
+
+	if (php_str2num(&num, ZSTR_VAL(numstr)) == FAILURE) {
+		zend_argument_value_error(1, "is not well-formed");
```

Comment:
I miss tests for the error cases (this one and a few more below)

Reply:
Thanks for checking!
I added value error case to tests.